dlp/api/DlpSample/InspectSamples.cs

```diff
@@ -421,89 +419,88 @@ internal class InspectSamples
+        };
 
-            var subscriptionName = new SubscriptionName(projectId, subscriptionId);
-            var subscriber = SubscriberClient.CreateAsync(subscriptionName).Result;
-            subscriber.StartAsync(
-                (pubSubMessage, cancellationToken) =>
-                {
-                    // Given a message that we receive on this subscription, we should either acknowledge or decline it
-                    if (pubSubMessage.Attributes["DlpJobName"] == dlpJob.Name)
-                    {
-                        fireEvent.Set();
-                        return Task.FromResult(SubscriberClient.Reply.Ack);
-                    }
+        // We need created job name
+        DlpJob dlpJob = client.CreateDlpJob(request);
 
-                    return Task.FromResult(SubscriberClient.Reply.Nack);
-                });
+        // Get a pub/sub subscription and listen for DLP results
+        ManualResetEventSlim fireEvent = new ManualResetEventSlim();
 
-            // We block here until receiving a signal from a separate thread that is waiting on a message indicating receiving a result of Dlp job
-            if (fireEvent.Wait(TimeSpan.FromMinutes(1)))
+        SubscriptionName subscriptionName = new SubscriptionName(projectId, subscriptionId);
+        SubscriberClient subscriber = SubscriberClient.CreateAsync(subscriptionName).Result;
+        subscriber.StartAsync(
+            (pubSubMessage, cancellationToken) =>
             {
-                // Stop the thread that is listening to messages as a result of StartAsync call earlier
-                subscriber.StopAsync(CancellationToken.None).Wait();
-
-                // Now we can inspect full job results
-                var job = client.GetDlpJob(new GetDlpJobRequest { DlpJobName = new DlpJobName(projectId, dlpJob.Name) });
-
-                // Inspect Job details
-                Console.WriteLine($"Processed bytes: {job.InspectDetails.Result.ProcessedBytes}");
-                Console.WriteLine($"Total estimated bytes: {job.InspectDetails.Result.TotalEstimatedBytes}");
-                var stats = job.InspectDetails.Result.InfoTypeStats;
-                Console.WriteLine("Found stats:");
-                foreach (var stat in stats)
+                // Given a message that we receive on this subscription, we should either acknowledge or decline it
+                if (pubSubMessage.Attributes["DlpJobName"] == dlpJob.Name)
                 {
-                    Console.WriteLine($"{stat.InfoType.Name}");
+                    fireEvent.Set();
+                    return Task.FromResult(SubscriberClient.Reply.Ack);
                 }
-            }
-            else
+
+                return Task.FromResult(SubscriberClient.Reply.Nack);
+            });
+
+        // We block here until receiving a signal from a separate thread that is waiting on a message indicating receiving a result of Dlp job
+        if (fireEvent.Wait(TimeSpan.FromMinutes(1)))
```

Comment:
Can you use a SemaphoreSlim instead so that you can await? Just wanting to show best practices in our samples.